data_steward/cdr_cleaner/cleaning_rules/null_invalid_foreign_keys.py

```diff
@@ -21,84 +29,215 @@
     'provider_id'
 ]
 
-INVALID_FOREIGN_KEY_QUERY = ('SELECT {cols} '
-                             'FROM `{project}.{dataset_id}.{table_name}` t '
-                             '{join_expr}')
+INVALID_FOREIGN_KEY_QUERY = JINJA_ENV.from_string("""
+    SELECT {{cols}}
+    FROM `{{project_id}}.{{dataset_id}}.{{table_name}}` t
+    {{join_expr}}""")
 
-LEFT_JOIN = ('LEFT JOIN `{dataset_id}.{table}` {prefix} '
-             'ON t.{field} = {prefix}.{field} ')
+LEFT_JOIN = JINJA_ENV.from_string("""
+    LEFT JOIN `{{dataset_id}}.{{table}}` {{prefix}} 
+    ON t.{{field}} = {{prefix}}.{{field}}""")
 
+SANDBOX_QUERY = JINJA_ENV.from_string("""
+    CREATE OR REPLACE TABLE
+    `{{project_id}}.{{sandbox_dataset_id}}.{{intermediary_table}}` AS (
+    SELECT {{cols}}
+    FROM `{{project_id}}.{{dataset_id}}.{{table_name}}` t
+    {{join_exp}})""")
```

Comment:
`SANDBOX_QUERY` is doing the same thing as `INVALID_FOREIGN_KEY_QUERY`. I think this cleaning rule is designed to update the records whose foreign keys are invalid, so we need a slightly different query to identify such records for sandboxing.

Reply:
@ChaoPang - yeah, this didn't seem right to me but it would be hard for me to test because this ticket doesn't include creating the integration test. Should I try to fix this now or do so in the integration test? @ksdkalluri what do you think I should do here? Fix now or fix during the creation of the integration test?

Reply:
Sandboxing query should be updated to get only the rows that are null when joined using a foreign key.

for example, we can use a similar query to this.

```
select m.* from measurement as m 
left join _mapping_visit_occurrence as mv using(visit_occurrence_id) 
where mv.visit_occurrence_id is NULL

```
to sandbox invalid visit_occurrence foreign keys in the measurement table.